### DISCUSSION TITLE
SBT remote caching & incremental compilation

### DISCUSSION THREAD

**Discussion by fhalde** [sbt/sbt#7296] [General]
Hello, we are unable to reap the benefits of remote caching and incremental compilation

SBT version: 1.4.9

**What we did?**

I tried compiling a project and did a `sbt "; pushRemoteCache" from my machine A
My team mate checked out the same branch and did a `sbt "; pullRemoteCache; compile;"

He received a cache miss

What else can I look at to debug why the cache was invalidated on his machine? AFAIK sbt started to use relative paths since 1.4.2 SBT version

Does remote cache require the dependencies to be freezed? What happens if I have a library version 1.2.0 and the other team member has the same dependency but with 1.2.1 version? I'm unsure if SBT does an auto-bump on dependencies

**Comment by SethTisue** [comment]
1.4.9 is quite old — is the problem reproducible on sbt 1.9.0?

**Comment by SethTisue** [comment]
> Does remote cache require the dependencies to be freezed?

It should. If it doesn't, I'd call that a bug.

(To be clear, I have no firsthand experience with remote caching. But if the inputs aren't the same, that should be a cache miss, and dependencies are very much inputs.)

**Comment by eed3si9n** [comment]
There are a few gotchas that Yoshida-san (@xuwei-k) had tracked - https://xuwei-k.hatenablog.com/entry/2020/10/29/004536. afaik he was able to work around most of those issues.
